pkg/codegen/java/templates.go

```diff
@@ -77,16 +78,28 @@ public class {{ .ClassName }} {
         return Optional.empty();
 	}
 ` + /* TODO: InvokeOptions probably should be done via a mutator on the InvokeOptions */ `
-	public static InvokeOptions withVersion(@Nullable InvokeOptions options) {
-            if (options != null && options.getVersion().isPresent()) {
-                return options;
-            }
-            return new InvokeOptions(
-                options == null ? null : options.getParent().orElse(null),
-                options == null ? null : options.getProvider().orElse(null),
-                getVersion()
-            );
+    public static InvokeOptions withVersion(@Nullable InvokeOptions options) {
+        if (options != null && options.getVersion().isPresent()) {
+            return options;
+        }
+        return new InvokeOptions(
+            options == null ? null : options.getParent().orElse(null),
+            options == null ? null : options.getProvider().orElse(null),
+            getVersion()
+        );
+    }
+
+    public static InvokeOutputOptions invokeOutputOptionsWithVersion(@Nullable InvokeOutputOptions options) {
```

Comment:
Not sure if it's a good idea, but you can overload static methods, so we could call this `withVersion` too -- terrible/won't work?